src/mesh/namebased_io.C

```diff
@@ -46,8 +46,10 @@
 #include <iomanip>
 #include <fstream>
 #include <vector>
-#include <sys/types.h> // getpid
-#include <unistd.h>
+#ifdef LIBMESH_HAVE_UNISTD_H
+#include <sys/types.h>
+#include <unistd.h>  // for getpid()
```

Comment:
This seems to only address half of the issue, since we still call `getpid()` elsewhere in this file. Does Windows have its own `getpid()` that requires no additional header inclusions to call? Or are we getting that inclusion by accident under Windows? According to some very brief investigating that I did, it looks like `getpid` is defined in [<process.h>](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getpid?view=msvc-170&viewFallbackFrom=vs-2019) on Windows.

Reply:
This is only a first set of changes for windows build.

Proper getpid(), gettimeofday(), mkstemp() definitions would require additional automake/autoconf-fu, so this will submitted separately.